packages/dev/core/src/assetContainer.ts

```diff
@@ -1050,4 +1050,61 @@ export class AssetContainer extends AbstractScene {
             }
         });
     }
+
+    /**
+     * @since
+     * Given a root asset, this method will traverse its hierarchy and add it, its children and any materials/skeletons/animation groups to the container.
+     * @param root
+     */
+    public addAllAssetsToContainer(root: Node) {
+        if (!root) {
+            return;
+        }
+
+        const nodesToVisit = new Array<Node>();
+        const visitedNodes = new Set<Node>();
+
+        nodesToVisit.push(root);
+
+        while (nodesToVisit.length > 0) {
+            const nodeToVisit = nodesToVisit.pop()!;
+            visitedNodes.add(nodeToVisit);

```

Comment:
Sorry, this is all because of Seb's OCD, but this feels like it should be at the end of the function!
So nit-picky from me....